### PR TITLE
Add operation_speed option to X9C component

### PR DESCRIPTION
**Pull request by oliverhihn**
# What does this implement/fix?

Added new configuration option `operation_speed` for the X9C digital potentiometer component. The `operation_speed` option allows users to control the delay between operations, providing more flexibility in adjusting the potentiometer's performance. The option can be set from 1 to 100 microseconds (1 being default as it was until now).

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** None

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):**  None

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
output:
  - platform: x9c
    id: my_x9c
    cs_pin: GPIO15
    inc_pin: GPIO16
    ud_pin: GPIO17
    initial_value: 0.5
    operation_speed: 50
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by ssieb** [comment]
"speed" makes it sound like bigger is faster.  Maybe use "delay" instead, since that's what it is.

**Comment by oliverhihn** [comment]
Yes, you are right. Maybe `step_delay` would be better suited?

**Comment by ssieb** [comment]
That's even better.

**Comment by oliverhihn** [comment]
I refactored it.

**Review by ssieb — APPROVED** [review]

File: esphome/components/x9c/output.py [hunk 1]
```diff
@@ -26,6 +27,7 @@
             cv.Optional(CONF_INITIAL_VALUE, default=1.0): cv.float_range(
                 min=0.01, max=1.0
             ),
+            cv.Optional(CONF_STEP_DELAY, default=1): cv.int_range(min=1, max=100),
```
Comment (jesserockz) [inline esphome/components/x9c/output.py:30]: This should be validated as a time period correctly
```suggestion
            cv.Optional(CONF_STEP_DELAY, default=1): cv.All(
                cv.positive_time_period_microseconds, 
                cv.Range(
                    min=cv.TimePeriod(microseconds=1), 
                    max=cv.TimePeriod(microseconds=100),
                )
            ),
```

Comment (oliverhihn) [reply esphome/components/x9c/output.py:30]: Should I create a new PR as this is already closed?

**Pull request merged**